core/domain/state_domain.py

```diff
@@ -1388,20 +1397,59 @@ def update_interaction_id(self, interaction_id):
         Args:
             interaction_id: str. The new interaction id to set.
         """
+        if interaction_id is None:
+            unwanted_content_ids = self._get_customization_args_content_ids()
+            self._update_content_ids_in_assets(unwanted_content_ids, [])
```

Comment:
I think it would be clearer to define self._delete_content_ids_in_assets() and have that call self._update_content_ids_in_assets() internally.

Reply:
sounds reasonable. I'll add this!